clar2wasm/src/words/conditionals.rs

```diff
@@ -525,12 +525,126 @@ impl Word for UnwrapErr {
                 consequent: unwrap_branch_id,
                 alternative: throw_branch_id,
             });
+
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+pub struct Try;
+
+impl Word for Try {
+    fn name(&self) -> ClarityName {
+        "try!".into()
+    }
+
+    fn traverse(
+        &self,
+        generator: &mut WasmGenerator,
+        builder: &mut walrus::InstrSeqBuilder,
+        _expr: &SymbolicExpression,
+        args: &[SymbolicExpression],
+    ) -> Result<(), GeneratorError> {
+        let input = args.get_expr(0)?;
+        generator.traverse_expr(builder, input)?;
+
+        let (succ_branch_id, throw_branch_id) = match generator.get_expr_type(input).cloned() {
+            Some(ref full_type @ TypeSignature::OptionalType(ref inner_type)) => {
+                let some_type = &**inner_type;
+
+                let some_locals = generator.save_to_locals(builder, some_type, true);
+
+                let mut throw_branch = builder.dangling_instr_seq(InstrSeqType::new(
+                    &mut generator.module.types,
+                    &[],
+                    &clar2wasm_ty(full_type),
+                ));
+
+                // in the case of throw, we need to restore the value, and re-push the discriminant
+                throw_branch.i32_const(0);
+                for local in &some_locals {
+                    throw_branch.local_get(*local);
+                }
```

Comment:
In this case, the value would just be placeholders, so it may be more efficient to push a placeholder to the stack using:
```
add_placeholder_for_clarity_type(throw_branch, some_type)
```
It's probably not much of a difference, if any, but it may be more clear to understand what's happening.

Reply:
done